Sources/MapboxNavigation/OrnamentsController.swift

```diff
@@ -70,12 +72,41 @@ extension NavigationMapView {
                                                    selector: #selector(orientationDidChange(_:)),
                                                    name: UIDevice.orientationDidChangeNotification,
                                                    object: nil)
+            NotificationCenter.default.addObserver(self,
+                                                   selector: #selector(didUpdateEHorizonRoadName),
+                                                   name: .electronicHorizonDidUpdatePosition,
+                                                   object: nil)
+
         }
         
         private func suspendNotifications() {
             NotificationCenter.default.removeObserver(self,
                                                       name: UIDevice.orientationDidChangeNotification,
                                                       object: nil)
+            NotificationCenter.default.removeObserver(self,
+                                                      name: .electronicHorizonDidUpdatePosition,
+                                                      object: nil)
+        }
+        
+        @objc func didUpdateEHorizonRoadName(_ notification: Notification) {
+            guard let startingEdge = notification.userInfo?[RoadGraph.NotificationUserInfoKey.treeKey] as? RoadGraph.Edge else { return }
+            let edgeIdentifier = startingEdge.identifier
+            
+            guard let routeController = navigationViewData.router as? RouteController,
+                  let metadata = routeController.roadGraph.edgeMetadata(edgeIdentifier: edgeIdentifier) else { return }
+            
+            electronicHorizonRoadNames = metadata.names.map { name -> String in
+                switch name {
+                case .name(let name):
+                    return name
+                case .code(let code):
+                    return "(\(code))"
+                }
+            }
+            
+            if electronicHorizonRoadNames.isEmpty {
+                electronicHorizonRoadNames = ["\(metadata.mapboxStreetsRoadClass.rawValue)"]
```

Comment:
This raw value could be something like `trunk`, uncapitalized and unlocalized, which wouldn’t be appropriate to show to the user and would be very unlikely to match a queried feature’s name.

Reply:
Thanks for the suggestion, I'll remove this part